### ISSUE TITLE
Migrations incompatible with sqlite: no such function: split_part

### ISSUE THREAD

**Issue by niklaskorz** [mautrix/whatsapp#749]
Upgrading from 0.10.x to 0.11.1 fails on sqlite as the `split_part` function used in the migrations seems to be specific to PostgreSQL and is not available in sqlite.

**Comment by tulir** [comment]
The bridge "polyfills" split_part so the migration works fine on sqlite

**Comment by niklaskorz** [comment]
Not on NixOS at least. After upgrading to 24.11, mautrix-whatsapp fails with the abovementioned error. @frederictobiasc are you seeing a similar issue?

**Comment by niklaskorz** [comment]
Logs:

```
Nov 30 21:51:05 korz-tech mautrix-whatsapp[2852]: Migrating legacy bridge config
Nov 30 21:51:05 korz-tech mautrix-whatsapp[2852]:   INF Initializing bridge built_at=0001-01-01T00:00:00Z go_version=go1.23.3 name=mautrix-whatsapp version=0.11.1+dev.unknown
Nov 30 21:51:05 korz-tech mautrix-whatsapp[2852]:   INF Detected legacy database, migrating... action="migrate legacy db"
Nov 30 21:51:05 korz-tech mautrix-whatsapp[2852]:   FTL Failed to migrate legacy database error="no such function: split_part" db_section=main
Nov 30 21:51:05 korz-tech systemd[1]: mautrix-whatsapp.service: Main process exited, code=exited, status=15/n/a
```

**Comment by tulir** [comment]
Ah, if you prevented the bridge from writing to the config, and use a legacy config that hasn't been updated in over 2 years, then it'll use the wrong database type and fail

**Comment by niklaskorz** [comment]
Yeah, that's a NixOS thing... I just saw that the config changes were not merged before 24.11 was released, so now mautrix-whatsapp will be in a broken state for a while for NixOS users: https://github.com/NixOS/nixpkgs/pull/350448

I'll see if I can help out getting the NixOS module changes in.

**Comment by frederictobiasc** [comment]
> Not on NixOS at least. After upgrading to 24.11, mautrix-whatsapp fails with the abovementioned error. @frederictobiasc are you seeing a similar issue?

No, with postgres everything continues to work. But I'm on nixos-unstable anyways.
